### PR TITLE
Fix broken timers when command buffer begin timestamp is missing.

### PR DESCRIPTION
**Pull request by florian-kuebler**
If we miss a command buffer's begin -- as we were not already
capturing -- we send "0" as special timestamp. However,
we use the first command buffer of a submission as a reference
to align the Gpu timestamps to our Cpu timeline. If this timestamp
was missing, we failed and resulted in timers with length of days.

This fixes this (for now) on the client site by discarding the
complete submission if the first command buffer has no begin
marker affected. Note that the "align" debug markers to the
begin of capture if their "begin" info is missing, remains working.

Follow up: We should re-consider sending incomplete submissions.

Test: Take a capture in the first actual frames of infiltrator.
Bug: http://b/181195585

### CODE REVIEW

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp [hunk 1]
```diff
@@ -159,6 +159,13 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuQueueSubmissionWit
   std::optional<GpuCommandBuffer> first_command_buffer =
       ExtractFirstCommandBuffer(gpu_queue_submission);
 
+  // The first command buffer acts as our reference needed to align GPU time based events in the
+  // CPU timeline. If we have no result for it -- which is the case if we started capturing within
+  // its execution -- we need to give up for this.
```
Comment (dpallotti) [inline src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:164]: Could you rephrase "give up for this"? What does "this" refer to? The command buffer (i.e., the queue submission)?

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp [hunk 2]
```diff
@@ -159,6 +159,13 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuQueueSubmissionWit
...
+  if (first_command_buffer.has_value() && first_command_buffer->begin_gpu_timestamp_ns() == 0) {
```
Comment (dpallotti) [inline src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:165]: What happens if `!first_command_buffer.has_value()`? Is that handled correctly in the calls that follow?

Comment (florian-kuebler) [reply src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:165]: Yes. The "first command buffer" is simply the first command buffer in this submission, if there are any. The code that follows always iterates of the command buffers, and does stuff for a concrete command buffer. Thus, in those cases the "first command buffer" is always set --  otherwise, we would just skip the loops.

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:164]: Done.

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp [hunk 3]
```diff
@@ -159,6 +159,13 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuQueueSubmissionWit
   std::optional<GpuCommandBuffer> first_command_buffer =
       ExtractFirstCommandBuffer(gpu_queue_submission);
 
+  // The first command buffer acts as our reference needed to align GPU time based events in the
+  // CPU timeline. If we have no result for it -- which is the case if we started capturing within
```
Comment (dpallotti) [inline src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:163]: Maybe change "result" to "initial GPU timestamp" (if I got it correctly)?

**Review by dpallotti — APPROVED** [review]

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp (hunk 3 above)
Comment (florian-kuebler) [reply src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp]: Somewhat. I changed it to "first timestamp of the submission"
